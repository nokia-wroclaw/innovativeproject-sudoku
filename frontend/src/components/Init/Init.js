import "./Init.css";
import React from "react";
import PropTypes from "prop-types";

function Init({ text }) {
  return (
    <div data-testid="container" className="center">
      <h1 data-testid="text">{text}</h1>
    </div>
  );
}

Init.propTypes = {
  text: PropTypes.string.isRequired
};

export default Init;
