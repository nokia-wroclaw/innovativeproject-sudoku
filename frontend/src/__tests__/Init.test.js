import React from "react";
import { render } from "@testing-library/react";
import Init from "../components/Init/Init";

it("renders", () => {
  const { asFragment } = render(<Init text="Sudoku Battle Royale" />);
  expect(asFragment()).toMatchSnapshot();
});

it("inserts text in h1", () => {
  const { queryByTestId } = render(<Init text="Sudoku Battle Royale" />);
  expect(queryByTestId("text")).toHaveTextContent("Sudoku Battle Royale");
});
