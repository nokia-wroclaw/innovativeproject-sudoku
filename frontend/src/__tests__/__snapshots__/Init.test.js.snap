// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="center"
    data-testid="container"
  >
    <h1
      data-testid="text"
    >
      Sudoku Battle Royale
    </h1>
  </div>
</DocumentFragment>
`;
